src/main/vcpf4/usb_conf.h

```diff
@@ -160,13 +160,21 @@
 
 /****************** USB OTG FS CONFIGURATION **********************************/
 #ifdef USB_OTG_FS_CORE
+#ifdef USB_CDC_HID
+ #define RX_FIFO_FS_SIZE                          128
+ #define TX0_FIFO_FS_SIZE                          32
+ #define TX1_FIFO_FS_SIZE                          32
+ #define TX2_FIFO_FS_SIZE                          32
```

Comment:
(probably ok, but why is TX2 and TX3 enabled? Only one IN endpoint was added?)

Reply:
Not really normal CDC uses 3 endpoints + HID (4th endpoint) and lastly USB controll endpoint EP0 which is 5 in total. I belive that BF doesn't use CDC Control endpoint 0x82

Reply:
Control messages are only received by BF  (baudrate + DTR)

Reply:
```Full Speed device @ 17 (0x14200000): .............................................   Communication device: "Worm FC"
    Port Information:   0x001a
           Not Captive
           Attached to Root Hub
           External Device
           Connected
           Enabled
    Number Of Endpoints (includes EP0):   
        Total Endpoints for Configuration 1 (current):   4
    Device Descriptor   
        Descriptor Version Number:   0x0200
        Device Class:   2   (Communication)
        Device Subclass:   2
        Device Protocol:   0
        Device MaxPacketSize:   64
        Device VendorID/ProductID:   0x0483/0x5740   (STMicroelectronics)
        Device Version Number:   0x0200
        Number of Configurations:   1
        Manufacturer String:   1 "Betaflight"
        Product String:   2 "Worm FC"
        Serial Number String:   3 "0x8000000"
    Configuration Descriptor (current config)   
        Length (and contents):   67
            Raw Descriptor (hex)    0000: 09 02 43 00 02 01 00 C0  32 09 04 00 00 01 02 02  
            Raw Descriptor (hex)    0010: 01 00 05 24 00 10 01 05  24 01 00 01 04 24 02 02  
            Raw Descriptor (hex)    0020: 05 24 06 00 01 07 05 82  03 08 00 FF 09 04 01 00  
            Raw Descriptor (hex)    0030: 02 0A 00 00 00 07 05 01  02 40 00 00 07 05 81 02  
            Raw Descriptor (hex)    0040: 40 00 00 
        Number of Interfaces:   2
        Configuration Value:   1
        Attributes:   0xC0 (self-powered)
        MaxPower:   100 mA
        Interface #0 - Communications-Control   
            Alternate Setting   0
            Number of Endpoints   1
            Interface Class:   2   (Communications-Control)
            Interface Subclass;   2
            Interface Protocol:   1
            Comm Class Header Functional Descriptor   
                Raw Descriptor (hex)   0000: 05 24 00 10 01 
            Comm Class Call Management Functional Descriptor   
                Raw Descriptor (hex)   0000: 05 24 01 00 01 
            Comm Class Abstract Control Management Functional Descriptor   
                Raw Descriptor (hex)   0000: 04 24 02 02 
            Comm Class Union Functional Descriptor   
                Raw Descriptor (hex)   0000: 05 24 06 00 01 
            Endpoint 0x82 - Interrupt Input   
                Address:   0x82  (IN)
                Attributes:   0x03  (Interrupt)
                Max Packet Size:   8
                Polling Interval:   255 ms
        Interface #1 - Communications-Data/Unknown Comm Class Model   
            Alternate Setting   0
            Number of Endpoints   2
            Interface Class:   10   (Communications-Data)
            Interface Subclass;   0   (Unknown Comm Class Model)
            Interface Protocol:   0
            Endpoint 0x01 - Bulk Output   
                Address:   0x01  (OUT)
                Attributes:   0x02  (Bulk)
                Max Packet Size:   64
                Polling Interval:   0 ms
            Endpoint 0x81 - Bulk Input   
                Address:   0x81  (IN)
                Attributes:   0x02  (Bulk)
                Max Packet Size:   64
                Polling Interval:   0 ms
```

It's hard for me to say why TX2 FIFO is 0 on BF if it's an used endpoint... It's possible that this endpoint is not really used and there is no data flowing to USB Host when just CDC is used. But I believe it's necessary to allocate some space when you use composite device but I will check it.

Reply:
Sorry, my knowledge is very limited here. IMO BF does not use 0x82 endpoint at all, but it is inherited from ST source.
Some configuration is supported, but EP0 is probably used